x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CrossClusterEsqlRCS2UnavailableRemotesIT.java

```diff
@@ -0,0 +1,310 @@
+            .setting("xpack.security.remote_cluster_server.ssl.enabled", "true")
+            .setting("xpack.security.remote_cluster_server.ssl.key", "remote-cluster.key")
+            .setting("xpack.security.remote_cluster_server.ssl.certificate", "remote-cluster.crt")
+            .setting("xpack.security.authc.token.enabled", "true")
+            .keystore("xpack.security.remote_cluster_server.ssl.secure_key_passphrase", "remote-cluster-password")
+            .node(0, spec -> spec.setting("remote_cluster_server.enabled", "true"))
+            .build();
+
+        queryCluster = ElasticsearchCluster.local()
+            .name("query-cluster")
+            .module("x-pack-esql")
+            .apply(commonClusterConfig)
+            .setting("xpack.ml.enabled", "false")
+            .setting("xpack.security.remote_cluster_client.ssl.enabled", "true")
+            .setting("xpack.security.remote_cluster_client.ssl.certificate_authorities", "remote-cluster-ca.crt")
+            .setting("xpack.security.authc.token.enabled", "true")
+            .keystore("cluster.remote.my_remote_cluster.credentials", () -> {
+                if (API_KEY_MAP_REF.get() == null) {
+                    final Map<String, Object> apiKeyMap = createCrossClusterAccessApiKey("""
+                        {
+                          "search": [
+                            {
+                                "names": ["*"]
+                            }
+                          ]
+                        }""");
+                    API_KEY_MAP_REF.set(apiKeyMap);
+                }
+                return (String) API_KEY_MAP_REF.get().get("encoded");
+            })
+            .rolesFile(Resource.fromClasspath("roles.yml"))
+            .user(REMOTE_METRIC_USER, PASS.toString(), "read_remote_shared_metrics", false)
+            .build();
+    }
+
+    @ClassRule
+    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
+
+    @Before
+    public void setupPreRequisites() throws Exception {
+        setupRolesAndPrivileges();
+        loadData();
+    }
+
+    public void testEsqlRcs2UnavailableRemoteScenarios() throws Exception {
+        clusterShutDownWithRandomSkipUnavailable();
+        remoteClusterShutdownWithSkipUnavailableTrue();
+        remoteClusterShutdownWithSkipUnavailableFalse();
+    }
+
+    private void clusterShutDownWithRandomSkipUnavailable() throws Exception {
+        // skip_unavailable is set to a random boolean value.
+        // However, no clusters are stopped. Hence, we do not expect any other behaviour
+        // other than a 200-OK.
+
+        configureRemoteCluster("my_remote_cluster", fulfillingCluster, false, randomBoolean(), randomBoolean());
+        String query = "FROM *,my_remote_cluster:* | LIMIT 10";
+        Response response = performRequestWithRemoteSearchUser(esqlRequest(query));
+
+        Map<String, Object> map = responseAsMap(response);
```

Comment:
nit: nothing wrong what is here, but something like
```
ObjectPath o = ObjectPath.createFromResponse(response)
assertThat(o.evaluate("_clusters.total"), is(2))
```
usually reads a bit nicer.